in_toto/models/link.py

```diff
@@ -85,22 +91,58 @@ class Link(Signable):
     _type = attr.ib()
     name = attr.ib()
     materials = attr.ib()
-    products = attr.ib()
     byproducts = attr.ib()
     command = attr.ib()
     environment = attr.ib()
 
     def __init__(self, **kwargs):
         super().__init__()
 
-        self._type = "link"
-        self.name = kwargs.get("name")
-        self.materials = kwargs.get("materials", {})
-        self.products = kwargs.get("products", {})
-        self.byproducts = kwargs.get("byproducts", {})
-        self.command = kwargs.get("command", [])
-        self.environment = kwargs.get("environment", {})
+        link = linkpb.Link()
+        link.name = kwargs.get("name", "")
```

Comment:
This is a breaking change, we should support the use of link predicates alongside legacy links for now.